lib/ruby_lsp/listeners/code_lens.rb

```diff
@@ -181,14 +181,15 @@ def add_test_code_lens(node, name:, command:, kind:)
         )
       end
 
-      sig { params(group_name: String, method_name: T.nilable(String)).returns(String) }
-      def generate_test_command(group_name:, method_name: nil)
+      sig { params(group_stack: T::Array[String], method_name: T.nilable(String)).returns(String) }
+      def generate_test_command(group_stack:, method_name: nil)
         command = BASE_COMMAND + T.must(@path)
+        group_name = T.must(group_stack.last)
```

Comment:
We also need to make this more precise for classes. I think that will require adding extra logic for the specs, which will continue to use a regex. Maybe we can introduce a new parameter for `spec_name`, which we use as a regex, and that will allow you to use `group_stack.join("::")` for running test classes.

Also, I we need to verify that it works the same for test-unit and update it as well.

Reply:
I think I found a pretty neat solution here. This should now work like you suggested but instead of just matching against `group_stack.join("::")` I also added a `\A` start of string matcher. This makes it even more precise in what is being executed. I did manual tests with this and its working as expected.